tareas/Tarea2/TareaVF2.jl

```diff
@@ -0,0 +1,674 @@
+cos(D::Dual)= Dual(cos(fun(D)),-sin(fun(D))*der(D))
+#-
+
+#-
+import Base: tan
+tan(D::Dual)= Dual(tan(fun(D)),der(D)*sec(fun(D)))
+#-
+
+#-
+import Base: asin
+asin(D::Dual)= Dual(asin(fun(D)),der(D)*(1/(1-fun(D)^2)^(1/2)))
+#-
+
+#-
+import Base: acos
+acos(D::Dual)= Dual(acos(fun(D)),der(D)*(-1/(1-fun(D)^2)^(1/2)))
+#-
+
+#-
+import Base: atan
+atan(D::Dual)= Dual(exp(atan(D)),der(D)*(1/(1+fun(D)^2)))
+#-
+
+#-
+import Base: sinh
+sinh(D::Dual)= Dual(sinh(fun(D)),der(D)*cosh(fun(D)))
+#-
+
+#-
+import Base: cosh
+cosh(D::Dual)= Dual(cosh(fun(D)),der(D)*sinh(fun(D)))
+#-
+
+#-
+import Base: tanh
+tanh(D::Dual)= Dual(tanh(fun(D)),der(D)*(1/(cosh(fun(d)))^2))
+#-
+
+#-
+import Base: asinh
+asinh(D::Dual)= Dual(asinh(fun(D)),der(D)*(1/(1+fun(D)^2)^(1/2)))
+#-
+
+#-
+import Base: acosh
+acosh(D::Dual)= Dual(acosh(fun(D)),der(D)*(1/(-1+fun(D)^2)^(1/2)))
+#-
+
+#-
+import Base: atanh
+atanh(D::Dual)= Dual(atanh(fun(D)),der(D)*(1/(1-fun(D)^2)))
+#-
+
+# ####Test's 
+
+#-
+#Raíz 
+ @testset "Raíz cuadrada de duales" begin 
+    @test sqrt(Dual(4, 1.0)) == Dual(2, 0.25)
+    @test sqrt(Dual(1,1)) == Dual(1, 0.5)
+end
+#-
+
+#-
+#Exponencial 
+@testset "Exponencial de Duales" begin 
+    @test exp(Dual(0,1)) == Dual(1, 1)
+    @test exp(Dual(2,3)) == Dual(exp(2), 3*exp(2))
+end
+#-
+
+#-
+#Logaritmo 
+@testset "Logaritmo de Duales" begin 
+    @test log(Dual(14,7)) == Dual(log(14), 2)
+    @test log(Dual(2,3)) == Dual(log(2), 2/3)
+end
+#-
+
+#-
+#Función sin  
+@testset "Función Sin de Duales" begin 
+    @test sin(Dual(0,7)) == Dual(0, 7)
+    @test sin(Dual(pi/2,1)) == Dual(1, cos(pi/2))
+end
+#-
+
+#-
+#Función cos 
+@testset "Función Cos de Duales" begin 
+    @test cos(Dual(0,2)) == Dual(1, 0)
+    @test cos(Dual(2*pi,2)) == Dual(1,-sin(2*pi)*2)
+end
+#-
+
+#-
+#Función tan  
+@testset "Función Tan de Duales" begin 
+    @test tan(Dual(1,0)) == Dual(tan(1),0)
+    @test tan(Dual(3,5)) == Dual(tan(3),5*sec(3))
+end
+#-
+
+#-
+#Función aSin   
+@testset "Función aSin de Duales" begin 
+    @test asin(Dual(1,-0.5)) == Dual(asin(1), -Inf )
+    @test asin(Dual(0.0,-0.5)) == Dual(0.0, -0.5)
+end
+#-
+
+#-
+#Función aCos  
+@testset "Función aCos de Duales" begin 
+    @test acos(Dual(0,1)) == Dual(acos(0), -1)
+    @test acos(Dual(1,2)) == Dual(0.0, -Inf)
+end
+#-
+
+#-
+#Función aTan 
+@testset "Función aTan de Duales" begin 
+    @test atan(Dual(1,0)) == Dual(atan(1), 0)
+    @test atan(Dual(1,2)) == Dual(atan(1), 1)
+end
+#-
+
+#-
+#Función sinh  
+@testset "Función Sinh de Duales" begin 
+    @test sinh(Dual(1,0)) == Dual(sinh(1), 0)
+    @test sinh(Dual(1,3)) == Dual(sinh(1), 3*cosh(1))
+end
+#- 
+
+#-
+#Función cosh  
+@testset "Función Cosh de Duales" begin 
+    @test cosh(Dual(1,0)) == Dual(cosh(1), 0)
+    @test cosh(Dual(1,3)) == Dual(cosh(1), 3*sinh(1))
+end
+#-
+
+#-
+#Función tanh 
+@testset "Función Tanh de Duales" begin 
+    @test tanh(Dual(0,2)) == Dual(0, 2)
+    @test tanh(Dual(1,2)) == Dual(tanh(1), 2*(1/(cosh(1)^2)))
```

Comment:
Los  tests  no pasan; el error dice "UndefVarError: d not defined"

Reply:
> Los tests no pasan; el error dice "UndefVarError: d not defined"

Ambos tests se corrigieron!!